MoviesApp/MoviesList/View/MoviesListViewController.swift

```diff
@@ -154,92 +150,32 @@ final class MoviesListViewController: UIViewController {
             }
         categoriesButtons.first?.backgroundColor = .darkGray
 
-        getMoviesPage(urlString: moviesCategories.getUrlString(page: nextPageNumber))
+        viewData = .loading
+        updateView()
     }
 
-    private func searchMovies(_ textField: UITextField, isPaginate: Bool) {
-        let page = isPaginate ? nextPageNumber : 1
-
-        if textField.hasText, let text = textField.text?.replacingOccurrences(of: " ", with: "%20") {
-            if !isPaginate { movies = [] }
-            getMoviesPage(urlString: Constants.getSearchMoviesURLString(page: page, searchedText: text))
-        } else {
-            if !isPaginate {
-                setCurrentCategoriesMovies()
-                return
+    private func fetchUpdates() {
+        let urlString = viewModel.getCurrentCategory().getUrlString(page: 1)
+        switch viewData {
+        case .loading:
+            viewModel.getMoviesPage(urlString)
+            updateView()
+        case let .data(movies):
+            viewModel.movies.append(contentsOf: movies)
+        case let .error(error):
+            showLoadingErrorAlert(title: "Ошибка!", message: error.localizedDescription) {
```

Comment:
```suggestion
            showLoadingErrorAlert(title: "Ошибка!", message: error.localizedDescription) {
```
в константу  "Ошибка!"

Reply:
```suggestion
            showLoadingErrorAlert(title: "Ошибка!", message: error.localizedDescription) { [weak self] in
```